jbpm/jbpm-flow/src/main/java/org/kie/kogito/process/impl/AbstractProcessInstance.java

```diff
@@ -851,13 +851,23 @@ public Throwable errorCause() {
             @Override
             public void retrigger() {
                 executeInWorkflowProcessInstanceWrite(pi -> {
+                    NodeInstance nodeInstanceInError = pi.getNodeInstance(nodeInstanceIdInError, true);
                     NodeInstanceImpl ni = (NodeInstanceImpl) pi.getByNodeDefinitionId(nodeInError, pi.getNodeContainer());
+
                     clearError(pi);
+
                     getProcessRuntime().getProcessEventSupport().fireProcessRetriggered(pi, pi.getKnowledgeRuntime());
                     org.kie.api.runtime.process.NodeInstanceContainer nodeInstanceContainer = ni.getNodeInstanceContainer();
                     if (nodeInstanceContainer instanceof NodeInstance) {
                         ((NodeInstance) nodeInstanceContainer).internalSetTriggerTime(new Date());
                     }
+
+                    if (Objects.nonNull(nodeInstanceInError) && Objects.isNull(nodeInstanceInError.getLeaveTime()) && Objects.isNull(nodeInstanceInError.getCancelType())) {
```

Comment:
```suggestion
                  if (nodeInstanceInError != null && nodeInstanceInError.getLeaveTime() == null && nodeInstanceInError.getCancelType() == null)
```
This is shorter and fastter. Objects.nonNull and Object.null are intended to be used as method reference for methods that accepts a predicate, not in regular java code

Reply:
ok! give me a sec